src/jvm/main/org/jetbrains/kotlinx/lincheck/trace/TracePoint.kt

```diff
@@ -323,8 +327,14 @@ internal class MethodCallTracePoint(
     }
 
     fun initializeParameters(parameters: List<String>, parameterTypes: List<String>) {
-        this.parameters = parameters
-        this.parameterTypes = parameterTypes
+        if (parameters.lastOrNull() == ObjectLabelFactory.CONTINUATION_REPRESENTATION) {
```

Comment:
1. Should we always apply this transformation? 
2. Should we do it here, or should it be a part of the post-processing?

Reply:
> Should we always apply this transformation?

If you are asking if this check always correctly identifies suspend functions, the answer is no. Technically any function can provide a continuation as last parameter. If we want to check this more thoroughly we might have to use [reflection](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.reflect/-k-function/is-suspend.html). But I am open to suggestions!

> Should we do it here, or should it be a part of the post-processing?

Not sure why not, this is the point when it becomes clear the method call is suspend. Which should be a property of the method call. Other option would be to only set the `isSuspendMethodCall` and based on that boolean remove the last parameter in a later stage which seems cumbersome.